packages/server/src/user/user.decorator.ts

```diff
@@ -0,0 +1,10 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import jwt_decode from 'jwt-decode';
+
+export const GetUser = createParamDecorator((data: unknown, ctx: ExecutionContext) => {
```

Comment:
The `JwtAuthGuard` typically will add the user to the context so when using the decorators you can avoid having to re-decode the JWT token.

For example [here](https://github.com/hicsail/object-storage-service/blob/main/packages/server/src/auth/user.decorator.ts) is a decorator I wrote that grabs the user from the context. I then can use that decorator as long as the endpoint is protected by the JWT Guard like [so](https://github.com/hicsail/object-storage-service/blob/main/packages/server/src/perms/perms.resolver.ts#L21). 

That way there is a bit better error handling and you don't have to keep parsing the JWT.